dot/state/service.go

```diff
@@ -7,28 +7,44 @@ import (
 	"fmt"
 	"math/big"
 	"path/filepath"
+	"time"
 
 	"github.com/ChainSafe/gossamer/dot/state/pruner"
 	"github.com/ChainSafe/gossamer/dot/telemetry"
 	"github.com/ChainSafe/gossamer/dot/types"
 	"github.com/ChainSafe/gossamer/internal/log"
+	"github.com/ChainSafe/gossamer/internal/metrics"
 	"github.com/ChainSafe/gossamer/lib/blocktree"
 	"github.com/ChainSafe/gossamer/lib/trie"
 	"github.com/ChainSafe/gossamer/lib/utils"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
 
 	"github.com/ChainSafe/chaindb"
 )
 
-const (
-	readyPoolTransactionsMetrics   = "gossamer/ready/pool/transaction/metrics"
-	readyPriorityQueueTransactions = "gossamer/ready/queue/transaction/metrics"
-	substrateNumberLeaves          = "gossamer/substrate_number_leaves/metrics"
-)
-
 var logger = log.NewFromGlobal(
 	log.AddContext("pkg", "state"),
 )
 
+var (
+	transactionPoolGauge = promauto.NewGauge(prometheus.GaugeOpts{
+		Namespace: "gossamer_state_transaction",
+		Name:      "pool_total",
+		Help:      "total number of transactions in ready pool",
+	})
+	transactionQueueGauge = promauto.NewGauge(prometheus.GaugeOpts{
+		Namespace: "gossamer_state_transaction",
+		Name:      "queue_total",
```

Comment:
If I'm not mistaken namespace should be the same for the whole program no?

Reply:
There's the auto included go metrics which are all prefixed with `go_metric_name`.  There's also system level metrics too.  I think I can add them to a registry that's already prefixed.